src/main/java/org/cloudfoundry/promregator/rewrite/CFAllLabelsMetricFamilySamplesEnricher.java

```diff
@@ -24,43 +26,39 @@
 	public static final String LABELNAME_INSTANCE_NUMBER = "cf_instance_number";
 	
 	private static String[] labelNames = new String[] { LABELNAME_ORGNAME, LABELNAME_SPACENAME, LABELNAME_APPNAME, LABELNAME_INSTANCEID, LABELNAME_INSTANCE_NUMBER };
-	
+	private Map<String, String> labelsAndValuesToAdd = new LinkedHashMap<>();
 	public static String[] getEnrichingLabelNames() {
 		return labelNames.clone();
 	}
-	
-	private String orgName;
-	private String spaceName;
-	private String appName;
-	private String instanceId;
 
 	public CFAllLabelsMetricFamilySamplesEnricher(String orgName, String spaceName, String appName, String instanceId) {
-		this.instanceId = instanceId;
-		this.spaceName = spaceName;
-		this.appName = appName;
-		this.orgName = orgName;
+		this.labelsAndValuesToAdd.put(LABELNAME_ORGNAME, orgName);
+		this.labelsAndValuesToAdd.put(LABELNAME_SPACENAME, spaceName);
+		this.labelsAndValuesToAdd.put(LABELNAME_APPNAME, appName);
+		this.labelsAndValuesToAdd.put(LABELNAME_INSTANCEID, instanceId);
+		this.labelsAndValuesToAdd.put(LABELNAME_INSTANCE_NUMBER, getInstanceFromInstanceId(instanceId));
 	}
 	
 	@Override
 	protected List<String> getEnrichedLabelNames(List<String> original) {
 		List<String> clone = new LinkedList<>(original);
-		Collections.addAll(clone, labelNames);
-		
+		Collections.addAll(clone, labelsAndValuesToAdd.keySet().toArray(new String[0]));
 		return clone;
 	}
 	
 	@Override
 	public List<String> getEnrichedLabelValues(List<String> original) {
 		List<String> clone = new LinkedList<>(original);
-		
-		clone.add(this.orgName);
-		clone.add(this.spaceName);
-		clone.add(this.appName);
-		clone.add(this.instanceId);
-		clone.add(getInstanceFromInstanceId(this.instanceId));
-		
+		Collections.addAll(clone, labelsAndValuesToAdd.values().toArray(new String[0]));
 		return clone;
 	}
+
+	@Override
+	protected void removeDuplicateLabels(List<String> original){
+		for(String label: original) {
```

Comment:
It would be good to log a warning if this behavior is occurring.

Reply:
@checketts 
However, if you get for every metric + each target + each scraping interval (default 15s!) an own logging entry, this could also be an operational nightmare.

Reply:
Perhaps make this logging switchable via configuration option, which is at state "log it" by default?

Reply:
Good point about the noise/quantity. Definitely would want to be able to mute it if it isn't of use.

In other codebases where this can occur, they just log it the first time at `WARN`, then at `DEBUG` for subsequent.

Reply:
I could also live with that approach. However, those "magic log level changes" can get on your nerves as well.

Let's ask our customer 😁 
@mcsammac: Which variant would you prefer as a user?

BTW: in this case this wouldn't be really easy to implement properly. We must expect that multiple threads could switch the "remember my decision" toggle here - and we would have to apply synchronization, if we really want to do it properly (however, if we would spool out two or three additional one's too much due to a race condition --- who would care?)

Reply:
I think i like the former of using a toggle.  Maybe could be tied to an overall logging level for the entire service?

Reply:
Sounds good to me

Reply:
Sounds good to me, too. 

@mcsammac: Will you give it a try or shall one of us do it?

Internal remark: General idea: create a special sublogger for https://github.com/promregator/promregator/blob/817d6410352c8ca764426544f8abceb4fac96b70/src/main/java/org/cloudfoundry/promregator/rewrite/CFAllLabelsMetricFamilySamplesEnricher.java. log with level `warn`. How to disable it then needs to be mentioned in the "logging" documentation.

Reply:
I can attempt to implement it, and go from there.